src/Savegame/BattleUnit.cpp

```diff
@@ -5119,6 +5119,26 @@ void getFactionScript(const BattleUnit *bu, int &faction)
 	faction = 0;
 }
 
+void hasItemScript(BattleUnit* bu, int &result, ScriptText itemName, int count)
```

Comment:
I think this should go to different PR, for function itself.
If we iterate for whole inventory and do lot of work why not return item it self?
Second, I would prefer having "strong" typed API, aka pass `const RuleItem*` not string.
This would require new function in mod that will convert string to rule object, but in long run this will be more correct version.

Reply:
Seperate PR: Will be done.
Return the BattleItem itself: Agreed.
Changing the argument to RuleItem: How would I get the RuleItem reference I want to search for? The use case is checking if a unit has a specific item in the inventory.

Reply:
You miss: "new function in mod that will convert string to rule object", we should start allowing scripts to access arbitrary unit and item rules objects. This will be building block for future functionalities, like create new item or new unit.

Reply:
Ah, I see. Yeah, I didn't quite read that right. Apologies.